src/Microsoft.Azure.EventHubs/Amqp/AmqpPartitionReceiver.cs

```diff
@@ -289,14 +289,16 @@ IList<AmqpDescribed> CreateFilters()
                     try
                     {
                         int batchSize;
+
                         lock (this.receivePumpLock)
                         {
                             if (this.receiveHandler == null)
                             {
                                 // Pump has been shutdown, nothing more to do.
                                 return;
                             }
-                            batchSize = receiveHandler.MaxBatchSize;
+
+                            batchSize = receiveHandler.MaxBatchSize != 0 ? receiveHandler.MaxBatchSize : ClientConstants.ReceiveHandlerDefaultBatchSize;
```

Comment:
Consider >= 0 instead of !=
That way if somebody sets a negative batch size, we still get a reasonable behavior.